scripts/security/tocve.py

```diff
@@ -0,0 +1,225 @@
+
+        for child in self.links:
+            if child["Type Name"] != "Sub-Task":
+                continue
+            child_id = child["To Issue"]
+            child_ticket = query_ticket(child_id)
+            child_links = query_weblinks(child_id)
+            links.extend([{
+                "ver": clean_version(child_ticket["Fix versions"]),
+                "URL": link["URL"],
+                } for link in child_links])
+
+        links.sort(key=lambda x: x["ver"])
+
+        return links
+
+    def get_fixed_versions(self):
+        return [x["ver"] for x in self.get_full_links()]
+
+def add_ref(refs, text):
+    refs.append({
+        "refsource": "CONFIRM",
+        "url": text})
+
+if __name__ == '__main__':
+    args = sys.argv[1:]
+    if len(args) != 2:
+        print("Usage: tocve.py ZEPSEC-nn {cve|rnote}")
+        sys.exit(1)
+    ticket = args[0]
+    command = args[1]
+
+    with open('cve-template.json') as fp:
+        cve = json.load(fp)
+    issue = Issue(ticket)
+    cve_id = issue.cve()
+
+    if command == 'cve':
+        aff_ver = issue.get_affected_versions()
+        valt = "This issue affects:\nzephyrproject-rtos zephyr"
+        for ver in aff_ver:
+            valt += "\nversion {} and later versions.".format(ver[1:])
+
+        cve["CVE_data_meta"]["ID"] = cve_id
+        cve["CVE_data_meta"]["TITLE"] = issue.summary()
+        cve["description"]["description_data"] = [{
+                "lang": "eng",
+                "value": issue.description() },
+                { "lang": "eng", "value": valt }
+                ]
+        cve["source"]["defect"] = [
+                url_base + ticket ]
+
+        refs = []
+        add_ref(refs, url_base + ticket)
+        add_ref(refs, vul_doc + "#" + cve_id)
+        for link in issue.get_links():
+            add_ref(refs, link)
+
+        cve["references"]["reference_data"] = refs
+
+        versions = []
+        for ver in aff_ver:
+            versions.append({
+                "version_affected": ">=",
+                "version_value": ver[1:]})
+        cve["affects"]["vendor"]["vendor_data"][0]["product"]["product_data"][0]["version"]["version_data"] = versions
+
+        json.dump(cve, sys.stdout, indent=4)
+
+    if command == 'rnote':
+        print(cve_id)
+        print('-' * len(cve_id))
+        print()
+        print(issue.summary())
+        print()
+        print(issue.description())
+        print()
+        aff = issue.get_fixed_versions()
+        aff[-1] = "and " + aff[-1]
+        print("This has been fixed in releases {}.".format(
+            ", ".join(aff)))
+        print()
+        print("- `CVS <{}#{}>`_".format(vul_doc, cve_id))
+        print()
+        print("- `Zephyr project bug tracker")
+        print("  <{}{}>`_".format(url_base, ticket))
+        print()
+        for link in issue.get_full_links():
+            print("- `PR fix for {}".format(link["ver"]))
+            print("  <{}>`_".format(link["URL"]))
+            print()
```

Comment:
Would be clear if you use a string template to do this

Reply:
You mean string interpolation, or f-Strings? String templates are also a python thing, but are more complex. I can use f-Strings, I just keep forgetting they are even there.

Reply:
f-strings would make it simpler